vir/src/data.rs

```diff
@@ -10,6 +10,7 @@ pub struct LocalData<'vir> {
     pub ty: Type<'vir>,
 }
 
+#[derive(Eq, PartialEq)]
```

Comment:
(Why were these needed?)

Reply:
It's necessary for [this function](https://github.com/zgrannan/prusti-dev/blob/9a110867ba73143343b6dd60bc0a864e8782e649/prusti-encoder/src/encoders/type/lifted.rs#L25-L40), which extracts the unique type parameters that should be used to instantiate the type, removing duplicate instances of the same parameter. For example, from type `Tuple3<T, U, Result<T, W>>` it would return `[T, U, W]`.